Failed to build using emcc when ENABLE_SDL=1 is set outside macOS

Error:
```
In file included from src/syscall_sdl.c:16:
In file included from /usr/include/SDL2/SDL.h:32:
In file included from /usr/include/SDL2/SDL_main.h:25:
In file included from /usr/include/SDL2/SDL_stdinc.h:31:
/usr/include/SDL2/SDL_config.h:4:10: fatal error: 'SDL2/_real_SDL_config.h' file not found
    4 | #include <SDL2/_real_SDL_config.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

The linker links system SDL library instead of Emscripten-port SDL library in commit 446b95b without properly checking which compiler is used.

--- a/Makefile
+++ b/Makefile
@@ -84,6 +84,7 @@ $(call set-feature, Zifencei)
 
 # Experimental SDL oriented system calls
 ENABLE_SDL ?= 1
+ifneq ("$(CC_IS_EMCC)", "1") # note that emcc generates port SDL headers/library, so it does not requires system SDL headers/library
 ifeq ($(call has, SDL), 1)
 ifeq (, $(shell which sdl2-config))
 $(warning No sdl2-config in $$PATH. Check SDL2 installation in advance)
@@ -100,6 +101,7 @@ OBJS_EXT += syscall_sdl.o
 $(OUT)/syscall_sdl.o: CFLAGS += $(shell sdl2-config --cflags)
 LDFLAGS += $(shell sdl2-config --libs) -pthread
 LDFLAGS += $(shell pkg-config --libs SDL2_mixer)
+endif
 endif
 
 ENABLE_GDBSTUB ?= 0
@@ -148,6 +150,8 @@ endif
 ifeq ("$(CC_IS_EMCC)", "1")
 ifeq ($(call has, SDL), 1)
 CFLAGS_emcc += -sUSE_SDL=2 -sSDL2_MIXER_FORMATS=wav,mid -sUSE_SDL_MIXER=2
+OBJS_EXT += syscall_sdl.o
+LDFLAGS += -pthread
 endif
 endif
 

